### PR TITLE
Feature: extra assembly annotations (sanntis)

### PR DESCRIPTION
**Pull request by SandyRogers**
This PR:
- adds "Assembly Extra Annotations" (model, views, urls, import script, tests/fixtures)
  - the idea of these is to support additional annotations, via GFFs, that are **not** part of the usual annotation pipelines, but are useful to show.
  - the initial use case is [SanntiS](https://github.com/Finn-Lab/SanntiS).
  - this is designed to be a bit generic, so that other tools could be supported in future.
  - it is also designed to be easily dropped, e.g. if SanntiS becomes part of pipeline v6.
  - these annots are related to `assemblies`, not to `analyses`, since they are essentially independent of the analysis run.
    - **only** `assemblies` are supported. In principle, other datatypes could be, but there wasn't a simple way to generalise that.
- drops Python 3.6 support
- adds support for (optional) `MGYG*_virify.gff` file during genomes import

### CODE REVIEW

File: emgapi/models.py [hunk 1]
```diff
@@ -114,7 +116,7 @@ def sync_with_ena_status(self, ena_model_status: ENAStatus):
 
         if ena_model_status == ENAStatus.DRAFT:
             logging.warning(
-                f"{study} will not be updated due to the study status being 'draft'"
+                f"{self} will not be updated due to the study status being 'draft'"
```
Comment (mberacochea) [inline emgapi/models.py:119]: Good catch hehe

File: emgapianns/management/commands/import_extra_assembly_annotations.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    ):
+        description_label, created = emg_models.DownloadDescriptionLabel \
+            .objects \
+            .get_or_create(description_label='SanntiS annotation', defaults={
+                "description": "SMBGC Annotation using Neural Networks Trained on Interpro Signatures"
+        })
+        if created:
+            logger.info(f'Added new download description label {description_label}')
+
+        fmt = emg_models.FileFormat \
```
Comment (mberacochea) [inline emgapianns/management/commands/import_extra_assembly_annotations.py:81]: Maybe we can cache this one?

File: tests/api/test_assemblies.py [hunk 3]
```diff
@@ -44,3 +42,15 @@ def test_asssemblies_list(self, client, assemblies, legacy_mapping):
...
+
+    def test_assembly_extra_annotations(self, client, assembly_extra_annotation):
+        assembly = assembly_extra_annotation.assembly
+
+        url = reverse("emgapi_v1:assembly-extra-annotations-list", args=[assembly.accession])
+        print(url)
+        response = client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+        rsp = response.json()
+        print(rsp)
```
Comment (mberacochea) [inline tests/api/test_assemblies.py:54]: a rogue print statement?

File: tests/test_utils/emg_fixtures.py [hunk 4]
```diff
@@ -622,17 +624,50 @@ def assemblies(study, runs, samples, experiment_type_assembly):
         accession="ERZ9999",
         study=study,
         experiment_type=experiment_type_assembly,
-        _quantity=10))
+        _quantity=1))
```
Comment (mberacochea) [inline tests/test_utils/emg_fixtures.py:627]: `1` instead of `10`?

**Review by mberacochea — APPROVED** [review]
Great stuff Sandy... I've added some really small comments.. I would say :shipit:

File: tests/test_utils/emg_fixtures.py (hunk 4 above)
Comment (SandyRogers) [reply tests/test_utils/emg_fixtures.py:627]: Yeh. I believe this was a typo... the comment above this is `# one with a fixed ERZ9999 to be used with the legacy mapping` and indeed it only makes sense to make 1 with a hardcoded accession. Doesn't seem to violate any DB uniqueness checks (probably should!), but I noticed it was wrong when doing a `.get` for this supposedly unique accession.

File: emgapianns/management/commands/import_extra_assembly_annotations.py (hunk 2 above)
Comment (SandyRogers) [reply emgapianns/management/commands/import_extra_assembly_annotations.py]: good shout, done

**Pull request merged**